### ISSUE TITLE
Clementine 1.3.1 is repeating some songs twice

### ISSUE THREAD

**Issue by devillemereuil** [clementine-player/Clementine#5489]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [ x] I checked the issue tracker for similar issues
- [x ] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [x ] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: openSUSE Tumbleweed
- Clementine version: 1.3.1

### Expected behaviour / actual behaviour
When playing a long playlist in "non-repeat" and "random: all" some songs (roughly half) are repeated twice. The expected behaviour would be that songs are only played once, and certainly not repeated immediately when "non-repeat" is selected.

### Steps to reproduce the problem (only for bugs)
- Have a long playlist, set it on "random:all" and make sure "repeat" is on "none/no repeat"
- Play songs
- Some songs start over immediately after ending, but only once (i.e. the song is played twice).

Can reproduce on every Clementine 1.3.1 I tested: openSUSE Leap/Tumbleweed and Kubuntu.

**Comment by nacenteno** [comment]
I have the same issue, on Ubuntu and Lubuntu 16.04, 32 and 64 bit

**Comment by pseth** [comment]
I too have the same issue on Ubuntu 16.04, 64 bit.

**Comment by devillemereuil** [comment]
Bug seems to be a duplicate of #5523. Can everyone confirm that WMA (or possibly AAC) songs are involved in triggering the bug? The problem is that the bug is not triggered when you select the file manually, so you have to wait for it to happen and check the track before the one that was played twice.

**Comment by niwics** [comment]
I can confirm that this annoying bug is related to WMA format. It appears when playing non-manually selected songs.
Ubuntu 16.04, 64 bit, Clementine 1.2.3

**Comment by gavelez** [comment]
I have the issue with mp3 and flac songs

**Comment by devillemereuil** [comment]
@gavelez Are you sure? Not that the song that is repeated can be of any format (including MP3), but the song **before** the repeated is always a WMA.

**Comment by MKevin3** [comment]
I have this same issue Clementine - V 1.3.1 Mac OS 10.12.3 (16D32)

I have a mix of MP3 and WMA files in the my playlist. Set to shuffle.

**Comment by freezelty** [comment]
1.3.1 on Mac 10.11.6. To me it is a WMA related bug. It happens as follow:

Step 1. Use Previous OR Next to jump to play a WMA song;
Step 2. When this WMA song (noted as Song A) finishes. the next song (noted as Song B) will be played but the "current track" highlighting is still on the just-finished WMA Song A;
Step 3. When Song B finishes the first time, the "current track" highlighting will then move from Song A to Song B, and start to play Song B again.

During the Step 2 above, if the seek bar is used to jump to the end of Song A, the "current track" highlighting will move to the next song correctly and thus the bug will not be triggered.

**Comment by Jeybla** [comment]
Thanks @freezelty that's a very accurate bug description. I was able to reproduce it:

![image](https://cloud.githubusercontent.com/assets/6203894/25351278/82d0770a-2928-11e7-97b1-c1bfd81ce5a5.png)
Using next:
![image](https://cloud.githubusercontent.com/assets/6203894/25351340/af64a49e-2928-11e7-901f-9257e1235016.png)
Waiting until the WMA track ends:
![image](https://cloud.githubusercontent.com/assets/6203894/25351407/e6584104-2928-11e7-9686-c354cc3c084d.png)
Currently `Iron Man - Black Sabbath (FLAC)` is already playing, but the interface shows still the WMA track.
![image](https://cloud.githubusercontent.com/assets/6203894/25351726/dd328f8e-2929-11e7-9692-d075f7d49d9f.png)
Iron Man is played twice. First with the old track displayed, and second with the correct track displayed. (By the way: ratings through a shortcut will also edit the wrong track)

So in short: `If you switch to a WMA track, the track after that will be repeated.`
I'll do the workaround and just convert my WMA files.

**Comment by HorlogeSkynet** [comment]
Same here (Debian 9, Clementine 1.3.1 too), for so long now. So glad I've finally found this issue 😄 
@freezelty, @Jeybla a big thanks to you. It's WMA-related, indeed 👍

But as mentionned in #5523, it happens only when the track is randomly played (not manually queued for instance).

**Comment by racuna** [comment]
Same here with .ogg files

**Comment by samweisgamdschie** [comment]
I have a list of .wma-only files.
* I click on a song (in the middle of the list) and it gets played one time
* although it starts playing the next song the first song is still marked as currently playing (blue warping).
* after the second song ends the second song gets played a second time and the marker now jumps to the second song.

It looks like there is something wrong with this marker for some reason.
Clementine: 1.2.3
Qt: 4.8.7
Linux Mint 18.2 Sonya (using the default ubuntu xenial repos in the background)

**Comment by blakejriley** [comment]
I have this issue with Clementine 1.2.3 on Ubuntu 16.04 64-bit when playing .wma files.

**Comment by Laurencio** [comment]
I have this issue with wma files on Clementine 1.3.1 on MX Linux 17.1.

**Comment by matkoniecz** [comment]
Sounds like duplicate of #3924

**Comment by divyanshvn** [comment]
Any updates on this ?

**Comment by RuudMulder** [comment]
Very strange. On OS X 10.13 I had this issue and noticed that the remaining time (bottom-right) was counting up instead of down. Clicking it to display total time and again got me a downcounter and the songs play only once.
